components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/TenantCreateGatewayObserver.java

```diff
@@ -0,0 +1,223 @@
+                            .getRegistry(RegistryType.SYSTEM_CONFIGURATION);
+
+            AxisConfiguration axisConfig = configurationContext.getAxisConfiguration();
+
+            // initialize the lock
+            Lock lock = new ReentrantLock();
+            axisConfig.addParameter("synapse.config.lock", lock);
+
+            // creates the synapse configuration directory hierarchy if not exists
+            // useful at the initial tenant creation
+            File tenantAxis2Repo = new File(configurationContext.getAxisConfiguration().getRepository().getFile());
+            File synapseConfigsDir = new File(tenantAxis2Repo, "synapse-configs");
+            if (!synapseConfigsDir.exists()) {
+                if (!synapseConfigsDir.mkdir()) {
+                    log.fatal("Couldn't create the synapse-config root on the file system for the tenant domain : "
+                                      + tenantDomain);
+                    return;
+                }
+            }
+
+            String synapseConfigsDirLocation = synapseConfigsDir.getAbsolutePath();
+            // set the required configuration parameters to initialize the ESB
+            axisConfig.addParameter(SynapseConstants.Axis2Param.SYNAPSE_CONFIG_LOCATION, synapseConfigsDirLocation);
+
+            // init the multiple configuration tracker
+            ConfigurationManager manger = new ConfigurationManager((UserRegistry) registry, configurationContext);
+            manger.init();
+
+            File synapseConfigDir = new File(synapseConfigsDir, manger.getTracker().getCurrentConfigurationName());
+            File buildSequenceFile = new File(synapseConfigsDir + File.separator + manger.getTracker()
+                    .getCurrentConfigurationName() + File.separator + MultiXMLConfigurationBuilder.SEQUENCES_DIR +
+                                                      File.separator + buildSequenceName + ".xml");
+
+            //Here we will check build sequence exist in synapse artifact. If it is not available we will create
+            //sequence synapse configurations by using resource artifacts
+            if (!buildSequenceFile.exists()) {
+                createTenantSynapseConfigHierarchy(synapseConfigDir, tenantDomain);
+            }
+        } catch (AxisFault e) {
+             log.error("Failed to create Tenant's synapse sequences for tenant " + tenantDomain);
+        } catch (ConfigurationInitilizerException e) {
+            log.error("Failed to create Tenant's synapse sequences for tenant. ");
+        }
+
+        try{
+            AppManagerUtil.loadTenantAPIPolicy(tenantDomain, tenantId);
+        }catch (AppManagementException e){
+            log.error("Failed to load tiers.xml to tenant's registry");
+        }
+    }
+
+
+    private ServerContextInformation initESB(String configurationName, ConfigurationContext configurationContext)
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/05d5cef1f9e3d525eb6d0c0af34943c96daac8ae